### DISCUSSION TITLE
Antd + Nextjs modules issue [ThemedLayoutContextProvider adds 6500 modules]!

### DISCUSSION THREAD

**Discussion by w20k** [refinedev/refine#4920] [Q&A] [answered]
Hi,

I have a project set up with antd, refine, and Next.js (scaffold: npm create refine-app@latest), and it begins with 6974 modules compiled during the initial load. I'm just wondering if there's a viable solution to improve its speed or if someone has discovered an option to address this issue.

![Screenshot 2023-09-07 at 16 53 26](https://github.com/refinedev/refine/assets/1053320/9064a06b-c939-4be1-b441-83ba2e9cbbdd)
![Screenshot 2023-09-07 at 16 50 59](https://github.com/refinedev/refine/assets/1053320/2a448cbd-9b55-4779-9794-9bb6ff6131c2)
![Screenshot 2023-09-07 at 16 51 18](https://github.com/refinedev/refine/assets/1053320/cc8676b1-63b0-4c41-8a58-5c7d91abccad)
**still looking for an issue.

Same setup, but already added few crud pages, but pretty much same thing as a scaffold.

So, had to manually copy/paste notificationProvider and change path for  RefineThemes. Now it's a bit better, 200ms instead of 2000ms :) 

<img width="1249" alt="Screenshot 2023-09-07 at 21 40 59" src="https://github.com/refinedev/refine/assets/1053320/c4de1f60-7c4a-479f-b4b5-096c028cff7b">
<img width="1293" alt="Screenshot 2023-09-07 at 21 40 45" src="https://github.com/refinedev/refine/assets/1053320/f766e6f5-3705-465f-8261-584963df7971">

``` 
//SIder, Header, Title and main ThemeLayout also should be copied
import { ThemedLayoutContextProvider } from '@refinedev/antd';
import { useThemedLayoutContext } from '@refinedev/antd';
///
import { ErrorComponent } from '@refinedev/antd';
....
import { notificationProvider } from '@refinedev/antd 
....
import { RefineThemes } from '@refinedev/antd';
```
After coping those files into the project - it helped drastically. Still kind of bad, but better then it was.

![Screenshot 2023-09-08 at 10 08 34](https://github.com/refinedev/refine/assets/1053320/5fc5f344-d276-4dc2-88a1-8a00a0629608)

node: v18.16.0
npm: 9.8.1
yarn: 1.22.19

**Comment by BatuhanW** [comment]
Hey @w20k, thanks for the report, we'll check it and get back to you.

**Comment by omeraplak** [comment]
Hey @w20k , apologies for the inconvenience. We suspect it's due to the ant-design/pro package. We're on it and will keep you posted

**Comment by w20k** [comment]
Are there any updates, or can you point me in the right direction for further research? I'm considering to give it a go and look at it. Working with antd and refine in development has become quite challenging.

**Comment by BatuhanW** [comment]
Hello @w20k, can you add the following line to your `next.config.js` file and let us know if that fixes your issue?

```
transpilePackages: [
    "@ant-design/icons-svg",
    "@refinedev/nextjs-router",
    "@refinedev/antd",
    "antd",
    "@ant-design/pro-components",
    "@ant-design/pro-layout",
    "@ant-design/pro-utils",
    "@ant-design/pro-provider",
    "rc-pagination",
    "rc-picker",
    "rc-util",
  ],
```

**Comment by w20k** [comment]
Hello, @BatuhanW, will do! Thank you!
I will test on my local project and demo to be sure if it's working or not!

**Reply by w20k** [reply]
Clean project:
<img width="853" alt="Screenshot 2024-01-22 at 20 21 23" src="https://github.com/refinedev/refine/assets/1053320/be7cea73-d7e5-45de-9f42-678b6aea47d1">

Local Pet Project:
<img width="861" alt="Screenshot 2024-01-22 at 20 25 19" src="https://github.com/refinedev/refine/assets/1053320/877c2a94-9c90-4e57-890c-0bd024e14469">

Pet project a little bit better, but still the same 😄

**Comment by BatuhanW** [comment, marked as answer]
Folks, I want to clarify that this issue isn't related to Refine, but it's an issue between NextJS and Ant Design. 

There are some open issues on nextjs side, which you can track for updates.

https://github.com/vercel/next.js/issues/48748

https://github.com/vercel/next.js/issues/5180

**Comment by w20k** [comment]
Thanks for your comment. Will use this 'discussion' to track those issues and update, if somehow they will resolve it 👍
